controllers/wavefront_controller_test.go

```diff
@@ -1103,86 +989,96 @@ func getEnvValueForName(envs []v1.EnvVar, name string) string {
 	require.NoError(t, err)
 
 	value := getEnvValueForName(deployment.Spec.Template.Spec.Containers[0].Env, "WAVEFRONT_PROXY_ARGS")
 	require.Contains(t, value, fmt.Sprintf("%s", proxyArg))
 }
 
-func setupScenario(wfCR *wf.Wavefront, initObjs ...runtime.Object) (*controllers.WavefrontReconciler, *wf.Wavefront, client.WithWatch, typedappsv1.AppsV1Interface) {
+func emptyScenario(wfCR *wf.Wavefront, initObjs ...runtime.Object) (*controllers.WavefrontReconciler, *testhelper.MockKubernetesManager) {
 	s := scheme.Scheme
-	s.AddKnownTypes(v1.SchemeGroupVersion, &v1.Service{})
-	s.AddKnownTypes(wf.GroupVersion, wfCR)
+	s.AddKnownTypes(wf.GroupVersion, &wf.Wavefront{})
 
 	clientBuilder := fake.NewClientBuilder()
-	clientBuilder = clientBuilder.WithScheme(s).WithObjects(wfCR)
+	if wfCR != nil {
+		clientBuilder = clientBuilder.WithScheme(s).WithObjects(wfCR)
+	}
 	clientBuilder = clientBuilder.WithScheme(s).WithRuntimeObjects(initObjs...)
-	apiClient := clientBuilder.Build()
-
-	initObjs = append(initObjs, &appsv1.Deployment{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       util.Deployment,
-			APIVersion: "apps/v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "wavefront-controller-manager",
-			Namespace: wfCR.Namespace,
-			UID:       "testUID",
-		},
-		Spec: appsv1.DeploymentSpec{
-			Template: v1.PodTemplateSpec{
-				Spec: v1.PodSpec{
-					Containers: []v1.Container{v1.Container{
-						Image: "projects.registry.vmware.com/tanzu_observability/kubernetes-operator:2.0.1",
-					}},
-				},
-			},
-		},
-	})
+	objClient := clientBuilder.Build()
 
-	initObjs = append(initObjs, &appsv1.Deployment{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       util.Deployment,
-			APIVersion: "apps/v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      util.ProxyName,
-			Namespace: wfCR.Namespace,
-		},
-		Status: appsv1.DeploymentStatus{
-			AvailableReplicas: 1,
-			Replicas:          1,
-		},
-	})
+	namespace := wftest.DefaultNamespace
+	if wfCR != nil {
+		namespace = wfCR.Namespace
+	}
+
+	if !containsObject(initObjs, operatorInNamespace(namespace)) {
+		operator := wftest.Operator()
+		operator.SetNamespace(namespace)
+		initObjs = append(initObjs, operator)
+	}
 
 	fakeClient := k8sfake.NewSimpleClientset(initObjs...)
 
+	mockKM := testhelper.NewMockKubernetesManager()
+
 	r := &controllers.WavefrontReconciler{
 		OperatorVersion:   "99.99.99",
-		Client:            apiClient,
+		Client:            objClient,
 		Scheme:            nil,
 		FS:                os.DirFS(controllers.DeployDir),
-		KubernetesManager: testhelper.NewMockKubernetesManager(),
+		KubernetesManager: mockKM,
 		MetricConnection:  metric.NewConnection(testhelper.StubSenderFactory(nil, nil)),
 		TypedClient:       fakeClient,
 	}
 
-	return r, wfCR, apiClient, fakeClient.AppsV1()
+	return r, mockKM
+}
+
+func componentScenario(wfCR *wf.Wavefront, initObjs ...runtime.Object) (*controllers.WavefrontReconciler, *testhelper.MockKubernetesManager) {
```

Comment:
The name `componentScenario` seemed to take a while to normalize for me. Juxtaposed with `emptyScenario` it makes more sense. I don't expect any action from this comment though, unless you can come up with a better name.

Reply:
I agree. Other names that I considered: `runningProxyScenario`, `subcomponentScenario`, `componentProvisioningScenario`,`dataCollectionScenario`/`collectionScenario` (since this is mostly used in cases where you want to test components used for data collection). `runningProxyScenario` was my runner up since that literally describes what it happening. However, it felt fragile.

Reply:
Let's leave it then and wait for the next iteration to change as needed.